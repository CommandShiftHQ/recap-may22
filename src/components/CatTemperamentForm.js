import React, { useState } from "react";

const formatTemperamentData = (temperamentData) => {
    return temperamentData.split(",").map(
        (temperament, index) => ({
            id: `${temperament}-${index}`,
            text: temperament,
            hasQuality: false
        })
    )
}


const CatTemperamentForm = ({ temperament }) => {
    const [formData, setFormData] = useState(formatTemperamentData(temperament))
    const [submitted, setSubmitted] = useState(false)

    const handleChange = (id) => {
        const newFormData = formData.map((item) => (
            item.id === id ? { ...item, hasQuality: !item.hasQuality } : item
        ))

        console.log({ newFormData })

        setFormData(newFormData)
    };

    const handleFormSubmit = (e) => {
        e.preventDefault();
        // axios.post somewhere
        setSubmitted(true);
    }

    return (
        <form
            onSubmit={handleFormSubmit}
        >
            {/* pass a name prop to this component and insert the cat's name below */}
            {submitted ? <p>Thank you!</p> : <><p>Would you agree that [insert cat's name] cats are: </p>
                {formData && formData.map((item, index) => (
                    <label key={index} data-testid="temperament-checkbox">
                        <input
                            type="checkbox"
                            value={item.text}
                            checked={item.hasQuality}
                            onChange={() => handleChange(item.id)}
                            id={item.id}
                        />

                        {item.text}

                    </label>))}
                <button>Submit</button>
            </>
            }

        </form>
    )
}

export default CatTemperamentForm;