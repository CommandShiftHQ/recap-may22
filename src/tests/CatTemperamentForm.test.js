import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';
import CatTemperamentForm from "../components/CatTemperamentForm"

describe("CatTemperamentForm", () => {
    const temperament = "Alert, Agile, Energetic, Demanding, Intelligent"

    it("renders as expected", () => {
        const { asFragment } = render(<CatTemperamentForm temperament={temperament} />);
        const component = asFragment();
        expect(component).toMatchSnapshot();
    });

    it("contains title", () => {
        render(<CatTemperamentForm temperament={temperament} />);
        expect(screen.getByText(/would you agree that/i)).toBeInTheDocument();
    });

    it("render expected number of cat qualities", () => {
        render(
            <CatTemperamentForm temperament={temperament} />
        );

        expect(screen.getAllByTestId("temperament-checkbox")).toHaveLength(5);
    });
});


