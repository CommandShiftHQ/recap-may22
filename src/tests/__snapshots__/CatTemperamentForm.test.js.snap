// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatTemperamentForm renders as expected 1`] = `
<DocumentFragment>
  <form>
    <p>
      Would you agree that ... cats are: 
    </p>
    <label
      data-testid="temperament-checkbox"
    >
      <input
        id="Alert-0"
        type="checkbox"
        value="Alert"
      />
      Alert
    </label>
    <label
      data-testid="temperament-checkbox"
    >
      <input
        id=" Agile-1"
        type="checkbox"
        value=" Agile"
      />
       Agile
    </label>
    <label
      data-testid="temperament-checkbox"
    >
      <input
        id=" Energetic-2"
        type="checkbox"
        value=" Energetic"
      />
       Energetic
    </label>
    <label
      data-testid="temperament-checkbox"
    >
      <input
        id=" Demanding-3"
        type="checkbox"
        value=" Demanding"
      />
       Demanding
    </label>
    <label
      data-testid="temperament-checkbox"
    >
      <input
        id=" Intelligent-4"
        type="checkbox"
        value=" Intelligent"
      />
       Intelligent
    </label>
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
